studio/components/layouts/DashboardLayout.tsx

```diff
@@ -0,0 +1,78 @@
+import Link from 'next/link'
+import IconBar from '../nav/IconBar'
+import Head from 'next/head'
+import { ReactElement } from 'react'
+import { Divider, Menu, Typography } from '@supabase/ui'
+
+type Sidebar = {
+  title: string
+  categories?: SidebarItems[]
+}
+type SidebarItems = {
+  label: string
+  links: SidebarLinks[]
+}
+type SidebarLinks = {
+  label: string
+  href: string
+}
+
+export default function DashboardLayout({
+  children,
+  title,
+  sidebar,
+}: {
+  children: ReactElement
+  title?: string
+  sidebar?: Sidebar
+}) {
+  return (
+    <>
+      <Head>
+        <title>Supabase Studio {title && `| ${title}`}</title>
+        <meta name="description" content="Supabase Studio" />
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+      <div className="flex">
+        <div className="w-14 h-screen bg-sidebar-light dark:bg-sidebar-dark border-r dark:border-dark">
+          <IconBar />
+        </div>
+        {sidebar && <SidebarMenu sidebar={sidebar} />}
+        <main className="flex-1">{children}</main>
+      </div>
+    </>
+  )
+}
+
+const SidebarMenu = ({ sidebar }: { sidebar: Sidebar }) => {
+  return (
+    <div
+      className={[
+        'w-64 h-screen overflow-autohide-scrollbar border-r ', // Layout
+        'bg-sidebar-linkbar-light', // Light mode
+        'dark:bg-sidebar-linkbar-dark dark:border-dark ', // Dark mode
+      ].join(' ')}
```

Comment:
Ahh this is fun!

Reply:
that was for my own sanity 😅  Perhaps will need to a refactor, but works for now!